packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -121,40 +193,157 @@ export function Tab<T>(props: TabProps<T>) {
             'is-selected': isSelected,
             'is-disabled': isDisabled,
             'is-hovered': isHovered
-          },
-          styleProps.className
+          }
         )}>
-        {icon}
-        {rendered && <span className={classNames(styles, 'spectrum-Tabs-itemLabel')}>{rendered}</span>}
+        <SlotProvider
+          slots={{
+            icon: {
+              size: 'S',
+              UNSAFE_className: classNames(styles, 'spectrum-Icon')
+            },
+            text: {
+              UNSAFE_className: classNames(styles, 'spectrum-Tabs-itemLabel')
+            }
+          }}>
+          {typeof rendered === 'string'
+            ? <Text>{rendered}</Text>
+            : rendered}
+        </SlotProvider>
       </div>
     </FocusRing>
   );
 }
 
-function TabLine({orientation, selectedTab}) {
+function TabLine(props) {
+  let {
+    orientation,
+    // Is either the tab node (non-collapsed) or the picker node (collapsed)
+    selectedTab,
+    // selectedKey is provided so that the TabLine styles are updated when the TabPicker's width updates from a selection change
+    selectedKey
+  } = props;
+
   let verticalSelectionIndicatorOffset = 12;
   let {direction} = useLocale();
   let {scale} = useProvider();
 
+  // TODO: Figure out the best way to get rid of the wiggle when it goes from non-collapsed to collapsed
+  // Only way I've found is to initialize the state with the calulated transform...
   let [style, setStyle] = useState({
     width: undefined,
     height: undefined,
-    transform: undefined
+    transform: `translateX(var(--spectrum-tabs-focus-ring-padding-x))`
```

Comment:
put this into the CSS file under the [dir=rtl] and [dir=ltr] selectors

Reply:
yep, this worked. Another thing for consideration:
Right now the tabline flies into position on initial render (try opening the default selected key story). I can get rid of this by performing the offset calculation outside the `useLayoutEffect` and setting it as the initial transform but it then introduces a slight wiggle when going from collapsed to non-collapsed :(